client/express-checkout/blocks/components/express-checkout-preview.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * External dependencies
+ */
+import { useState } from 'react';
+import { Elements, ExpressCheckoutElement } from '@stripe/react-stripe-js';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import InlineNotice from 'components/inline-notice';
+
+export const ExpressCheckoutPreviewComponent = ( {
+	stripe,
+	buttonType,
+	theme,
+	height,
+} ) => {
+	const [ canRenderButtons, setCanRenderButtons ] = useState( true );
+
+	const options = {
+		mode: 'payment',
+		amount: 1000,
+		currency: 'usd',
+	};
+
+	const mapThemeConfigToButtonTheme = ( paymentMethod, buttonTheme ) => {
+		switch ( buttonTheme ) {
+			case 'dark':
+				return 'black';
+			case 'light':
+				return 'white';
+			case 'light-outline':
+				if ( paymentMethod === 'googlePay' ) {
+					return 'white';
+				}
+
+				return 'white-outline';
+			default:
+				return 'black';
+		}
+	};
+
+	const googlePayType = buttonType === 'default' ? 'plain' : buttonType;
+
+	const applePayType = buttonType === 'default' ? 'plain' : buttonType;
+
+	let shouldOnlyShowApplePay = false;
+	if ( window.ApplePaySession && window.ApplePaySession.canMakePayments() ) {
+		shouldOnlyShowApplePay = true;
+	}
```

Comment:
I don't think this method of checking if we should show Apple Pay is fool-proof. Apple says in [their documentation](https://developer.apple.com/documentation/apple_pay_on_the_web/apple_pay_js_api/checking_for_apple_pay_availability):

> <dl>
> <dt><a href="https://developer.apple.com/documentation/apple_pay_on_the_web/applepaysession/1778027-canmakepayments">canMakePayments</a></dt>
> <dd>This method verifies that the device is capable of making Apple Pay payments; it doesn’t verify that the person has a provisioned card for use with Apple Pay on the device. You can call this method at any time.</dd>
> </dl>

I think that means it's possible for `canMakePayments()` to return `true` even if the Apple Pay button can't actually be displayed?

If that happens the info notice should be triggered and let the merchant know they need to do to render the button in Safari, but it'd be better if we could just show Google Pay if they're logged into that instead?

And if we choose to always show both buttons this isn't even a problem we need to think about.

Reply:
>I think that means it's possible for canMakePayments() to return true even if the Apple Pay button can't actually be displayed?

That's my understanding as well.

>And if we choose to always show both buttons this isn't even a problem we need to think about.

FWIW, I think we should be displaying both whenever possible since we're just displaying a preview of them. I saw you asked the design about that so let's wait for their decision 👀

Reply:
Displaying both as of f05a885669b79915f7e7e9f058a6b8241f28b11e per https://github.com/Automattic/woocommerce-payments/pull/9028#issuecomment-2211295785